core/backend/src/ECSchemaXmlContext.ts

```diff
@@ -25,23 +26,38 @@ export class ECSchemaXmlContext {
   }
 
   public addSchemaPath(searchPath: string): void {
-    this._nativeContext!.addSchemaPath(searchPath);
+    if (undefined === this._nativeContext) {
+      throw new IModelError(BentleyStatus.ERROR, "Native context is undefined.");
+    }
+    this._nativeContext.addSchemaPath(searchPath);
   }
 
   public setSchemaLocater(locater: IModelJsNative.ECSchemaXmlContext.SchemaLocaterCallback): void {
-    this._nativeContext!.setSchemaLocater(locater);
+    if (undefined === this._nativeContext) {
+      throw new IModelError(BentleyStatus.ERROR, "Native context is undefined.");
+    }
+    this._nativeContext.setSchemaLocater(locater);
   }
 
   public setFirstSchemaLocater(locater: IModelJsNative.ECSchemaXmlContext.SchemaLocaterCallback): void {
-    this._nativeContext!.setFirstSchemaLocater(locater);
+    if (undefined === this._nativeContext) {
+      throw new IModelError(BentleyStatus.ERROR, "Native context is undefined.");
+    }
+    this._nativeContext.setFirstSchemaLocater(locater);
   }
 
   public readSchemaFromXmlFile(filePath: string): any {
-    const response = this._nativeContext!.readSchemaFromXmlFile(filePath);
+    if (undefined === this._nativeContext) {
```

Comment:
these exceptions are useless. The program is going to crash either way.

Reply:
They aren't exactly the same in both cases. The error you get with using an undefined variable is mostly useless for tracking down issues (the call stack doesn't even have sourcemaps setup so it's not useful).

Whereas with an error thrown it can provide a more descriptive error about what has gone wrong.